rust/src/detect/flow.rs

```diff
@@ -0,0 +1,188 @@
+    let (i, du32) = detect_parse_uint::<u32>(i)?;
+    return Ok((
+        i,
+        DetectFlowPkts {
+            pkt_data: du32,
+            dir: dir,
+        },
+    ));
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn SCDetectFlowPktsParseDir(
+    ustr: *const std::os::raw::c_char, dir: DetectFlowDir,
+) -> *mut DetectFlowPkts {
+    let ft_name: &CStr = CStr::from_ptr(ustr); //unsafe
+    if let Ok(s) = ft_name.to_str() {
+        if let Ok((_, ctx)) = detect_parse_flow_pkts_dir(s, dir) {
+            let boxed = Box::new(ctx);
+            return Box::into_raw(boxed) as *mut _;
+        }
+    }
+    return std::ptr::null_mut();
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn SCDetectFlowPktsParse(
+    ustr: *const std::os::raw::c_char,
+) -> *mut DetectFlowPkts {
+    let ft_name: &CStr = CStr::from_ptr(ustr); //unsafe
+    if let Ok(s) = ft_name.to_str() {
+        if let Ok((_, ctx)) = detect_parse_flow_pkts(s) {
+            let boxed = Box::new(ctx);
+            return Box::into_raw(boxed) as *mut _;
+        }
+    }
+    return std::ptr::null_mut();
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn SCDetectFlowPktsFree(ctx: &mut DetectFlowPkts) {
+    // Just unbox...
+    std::mem::drop(Box::from_raw(ctx));
+}
+
+pub fn detect_parse_flow_bytes(i: &str) -> IResult<&str, DetectFlowBytes> {
+    let (i, _) = opt(is_a(" \t"))(i)?;
+    let (i, fd) = detect_parse_flow_direction(i)?;
+    let (i, _) = opt(is_a(" \t"))(i)?;
+    let (i, _) = tag(",")(i)?;
+    let (i, _) = opt(is_a(" \t"))(i)?;
+    return detect_parse_flow_bytes_dir(i, fd);
+}
+
+pub fn detect_parse_flow_bytes_dir(i: &str, dir: DetectFlowDir) -> IResult<&str, DetectFlowBytes> {
+    let (i, du64) = detect_parse_uint::<u64>(i)?;
+    return Ok((
+        i,
+        DetectFlowBytes {
+            byte_data: du64,
+            dir: dir,
+        },
+    ));
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn SCDetectFlowBytesParseDir(
+    ustr: *const std::os::raw::c_char, dir: DetectFlowDir,
+) -> *mut DetectFlowBytes {
+    let ft_name: &CStr = CStr::from_ptr(ustr); //unsafe
+    if let Ok(s) = ft_name.to_str() {
+        if let Ok((_, ctx)) = detect_parse_flow_bytes_dir(s, dir) {
+            let boxed = Box::new(ctx);
+            return Box::into_raw(boxed) as *mut _;
+        }
+    }
+    return std::ptr::null_mut();
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn SCDetectFlowBytesParse(
+    ustr: *const std::os::raw::c_char,
+) -> *mut DetectFlowBytes {
+    let ft_name: &CStr = CStr::from_ptr(ustr); //unsafe
+    if let Ok(s) = ft_name.to_str() {
+        if let Ok((_, ctx)) = detect_parse_flow_bytes(s) {
+            let boxed = Box::new(ctx);
+            return Box::into_raw(boxed) as *mut _;
+        }
+    }
+    return std::ptr::null_mut();
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn SCDetectFlowBytesFree(ctx: &mut DetectFlowBytes) {
+    // Just unbox...
+    std::mem::drop(Box::from_raw(ctx));
```

Comment:
I think we can remove this comment for now.  `Just unbox...` is really just:
```
let _ = Box::from_raw(ctx);
```
They both effectively do the same thing  tho.  But this code is unbox with explicit drop.

Reply:
Its a comment we've carried over in error lots before :)

Reply:
ok for removing indeed